File: JZAndroid/JavaZoneApiClient/src/main/java/no/schedule/javazone/v3/sync/FeedbackApiService.java
package no.schedule.javazone.v3.sync;

import com.google.gson.JsonElement;

import no.schedule.javazone.v3.io.model.Feedback;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.http.Body;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface FeedbackApiService {
  @Headers( "Content-Type: application/json" )
  @POST("/events/{eventId}/sessions/{sessionId}/feedbacks")
  Call<Object> postSessionFeedback(@Path("eventId") String eventId,
                                   @Path("sessionId") String sessionId,
                                   @Header("Voter-ID") String voterId,
                                   @Body Feedback jzFeedbackBody);

  @Headers("Content-Type: application/json")
  @POST("/events/58b3bfaa-4981-11e5-a151-feff819cdc9f/sessions/58b3c298-4981-11e5-a151-feff819cdc9f/feedbacks")
  Call<String> postSessionFeedbackTest(@Header("Voter-ID") String voterId,
                                       @Body Feedback jzFeedbackBody,
                                       Callback<JsonElement> success);

}

File: JavaZoneiOS/JavaZone/Sync/FeedbackApiService.swift
import Foundation
import Alamofire
import PromiseKit
import SVProgressHUD

class FeedbackApiService {
    static let sharedInstance = FeedbackApiService()
    private var manager: SessionManager
    
    private init() {
        self.manager = Alamofire.SessionManager.default
    }
    
    public func generateUniqueDeviceId() -> String? {
        let deviceId = UIDevice.current.identifierForVendor!.uuidString
        return deviceId.toBase64()
    }
    
    public func submitFeedback(session: Session!, feedback: Feedback) -> Promise<String> {
        let eventId = session!.conferenceId!
        let sessionId = session!.sessionId!
        let url = "\(JZURL.GetDevNullUrl)/events/\(eventId)/sessions/\(sessionId)/feedbacks"
        let parameters = feedback.toJSON()
        
        let headers: HTTPHeaders = [
            "Voter-ID": "\(generateUniqueDeviceId()!.toBase64())",
            "Content-Type": "application/json",
            "Accept": "application/json"]
        
        SVProgressHUD.show()
        return Promise<String> { seal in
            self.manager.request(url, method: .post, parameters: parameters, encoding: JSONEncoding.default, headers: headers).validate(statusCode: 200..<300).responseJSON { response in
                switch response.result {
                case .success:
                    //to get JSON return value
                    guard (response.result.value as? AnyObject) != nil else {
                        seal.reject(NSError(domain: "domain", code: 0, userInfo: [NSLocalizedDescriptionKey: "Some error reading response"]))
                        return
                    }
                    
                    seal.fulfill(response.result.description)
                case .failure(let error):
                    seal.reject(error)
                }
            }
        }
    }
}
